src/createBrowserClient.ts

```diff
@@ -30,180 +60,52 @@ export function createBrowserClient<
   supabaseUrl: string,
   supabaseKey: string,
   options?: SupabaseClientOptions<SchemaName> & {
-    cookies?: CookieMethods;
+    cookies?: CookieMethodsBrowser;
     cookieOptions?: CookieOptionsWithName;
     isSingleton?: boolean;
   },
 ) {
+  if ((options?.isSingleton || isBrowser()) && cachedBrowserClient) {
```

Comment:
Let's go with this because to me it makes most sense. If I set the `isSingleton` is not set, then it should only apply in a browser. If it's explicitly set then it should be obeyed.